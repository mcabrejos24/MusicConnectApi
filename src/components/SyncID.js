import { useEffect, useState } from "react";
import "../assets/styles/components/sync-id.scss";

export default function MusicCard2(props) {
  const { syncReady } = props;

  const [syncID, setSyncID] = useState(false);

  let filter = [];
  const keypadZero = 48;
  for (let i = 0; i <= 9; i++) {
    filter.push(i + keypadZero);
  }
  filter.push("Backspace");
  filter.push("ArrowLeft");
  filter.push("ArrowRight");
  filter.push("Del");
  filter.push("Meta");

  useEffect(() => {
    let syncIDButton = document.querySelector(`.sync-id-submit-button`);

    if (syncID) {
      syncIDButton.disabled = false;
    } else {
      syncIDButton.disabled = true;
    }
  }, [syncID]);

  function checkInput(event) {
    // onKeyDown
    const target = event.target;
    if (!target || !target.type) {
      console.error(
        "Error: could not find target or target type from syncID input."
      );
      return;
    }

    if (target.type === "tel") {
      if (
        filter.indexOf(event.key.charCodeAt(0)) < 0 &&
        filter.indexOf(event.key) < 0
      ) {
        event.preventDefault();
        alert("Whoops that's not a digit! please enter a digit to continue."); // tell user to enter a number
      }
    }
  }

  function formatSyncID({ target }) {
    if (!target) {
      console.error(
        "Error: could not find target or target value from syncID input."
      );
      return;
    }

    if (target.value.length > 7) {
      // stops taking input at 6 characters
      target.value = target.value.slice(0, -1);
      return;
    }
    target.value = target.value.replaceAll("-", "");

    if (target.value.length > 3) {
      target.value = target.value.slice(0, 3) + "-" + target.value.slice(3);
    }

    validateSync(target.value);
  }

  function validateSync(value) {
    let syncIdInput = document.querySelector(`.sync-id-input`);

    if (!value) {
      // resets style
      syncIdInput.classList.remove("input-contains");
      syncIdInput.classList.remove("input-does-not-contain");
      setSyncID(false);
      return;
    }
    const inputIsNum = /\d{3}[-]?\d{3}/.test(value); // check length and make sure it matches type
    setSyncID(inputIsNum);
    if (inputIsNum) {
      syncIdInput.classList.add("input-contains");
      if (syncIdInput.classList.contains("input-does-not-contain"))
        syncIdInput.classList.remove("input-does-not-contain");
    } else {
      syncIdInput.classList.add("input-does-not-contain");
      if (syncIdInput.classList.contains("input-contains"))
        syncIdInput.classList.remove("input-contains");
    }
  }

  function submitSyncID() {
    const syncIdInput = document.querySelector(`.sync-id-input`)?.value;
    if (!syncIdInput) {
      console.error("Cannot find syncID value.");
      syncReady(false);
      return;
    }

    const syncValid = "spotify"; // send syncID to back end and get back some value, if that value is true then continue
    if (!syncValid)
      alert(
        "Sync ID you provided does not exist. Please go back and use one of the other two options."
      );
    syncReady(syncValid);
  }

  return (
    <div className="sync-id">
      <p className="sync-id-description">Enter your Sync ID below</p>
      <input
        onKeyDown={checkInput}
        onKeyUp={formatSyncID}
        type="tel"
        className="sync-id-input"
        placeholder="ex: 123-456"
      />
      <button className="sync-id-submit-button" onClick={submitSyncID}>
        Submit
      </button>
    </div>
  );
}
