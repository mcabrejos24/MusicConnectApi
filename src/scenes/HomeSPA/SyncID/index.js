import { useState } from "react";
import BackButton from "../../../components/BackButton";
import MusicCard3 from "../../../components/MusicCard3";
import AuthorizerAppleMusic from "../../../components/AuthorizerAppleMusic";
import AuthorizerSpotify from "../../../components/AuthorizerSpotify";
import SubmitSync from "../../../components/SubmitSync";

export default function SyncID() {
  const [playlistReady, setPlaylistReady] = useState(false);

  function openPopUp(pop) {
    const popup = document.getElementById("popup");
    popup.style.visibility = "visible";
    popup.style.opacity = "1";
  }

  function closePopUp() {
    const popup = document.getElementById("popup");
    popup.style.visibility = "hidden";
    popup.style.opacity = "0";
  }

  setTimeout(openPopUp,500);

  return (
    <div className="content">

      <div className="coming-soon">
        <div id="popup" className="popup">
          <div className="popup-card">
            <h2>Coming Soon!</h2>
            <button id="close" class="close" onClick={closePopUp}>&times;</button>
            <p className="popup-card--content">This page is still in development so submitting won't actually do anything. But feel free to mess around!</p>
          </div>
        </div>
      </div>

      <BackButton></BackButton>
      <div className="double-authorizer-wrapper">
        <MusicCard3 confirmPlaylist={setPlaylistReady}>
          <AuthorizerAppleMusic />
          <AuthorizerSpotify />
        </MusicCard3>
      </div>
      <SubmitSync playlistReady={playlistReady} />

            <div className="how-it-works">
                <h1 className="how-it-works-title">How to use this sync ID option</h1>
                <div className="point need">
                    <h2 className="point-title">Submit Sync ID</h2>
                    <div className="point-description">
                        <p>This one is made for when both your peer has already done the second option.
                            <br /><br />
                            Simply, enter the Sync ID that was emailed or texted to you.
                            <br /><br />
                            You then choose the playlist you would like to sync.
                            <br /><br />
                            If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
                            <br /><br />
                            Then, since you peer has already provided their information, you just need to press the Sync button and your playlists will be synced!
                            <br /><br />
                            Enjoy sharing your playlist!
                        </p>
                    </div>
                </div>
            </div>
      
    </div>
  );
}
