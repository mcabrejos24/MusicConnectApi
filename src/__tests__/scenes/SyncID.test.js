import React from "react";
import { shallow } from "enzyme";
import SyncID from '../../scenes/HomeSPA/SyncID';
import toJson from "enzyme-to-json";

describe("renders SyncID page", () => {
    let page;
    beforeEach(() => {
        page = shallow(<SyncID />);
    });
    it("SyncID loads", () => {
        page;
    });
    it("props class name matches", () => {
        expect(page.props().className).toEqual("content sync-id-page");
    });
    it("matches snapshot", () => {
        expect(toJson(page)).toMatchSnapshot();
    });
    it("pop up renders", () => {
        // setTimeout(() => {}),500);
        expect(page.find(".popup").text()).toEqual("Coming Soon!Ã—This page is still in development so submitting won't actually do anything. But feel free to mess around!");
    });
    it("pop up renders", () => {
        expect(page.find(".popup").hasClass("popup-show")).toEqual(true);
        // expect(page.find(".popup").hasClass("popup-hide")).toEqual(false);
    });
    it("can dismiss popup", () => {
        page.find(".popup").simulate("click")
        // expect(page.find(".popup").hasClass("popup-hide")).toEqual(false);
    });
});


    // const popup = document.getElementById("popup");
    // popup.style.visibility = "visible";
    // popup.style.opacity = "1";