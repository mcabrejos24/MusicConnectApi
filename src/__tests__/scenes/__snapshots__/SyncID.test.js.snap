// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SyncID page matches snapshot 1`] = `
<div
  className="content sync-id-page"
>
  <div
    className="coming-soon"
  >
    <div
      className="popup"
      id="popup"
    >
      <div
        className="popup-card"
      >
        <h2>
          Coming Soon!
        </h2>
        <button
          className="close"
          id="close"
          onClick={[Function]}
        >
          Ã—
        </button>
        <p
          className="popup-card--content"
        >
          This page is still in development so submitting won't actually do anything. But feel free to mess around!
        </p>
      </div>
    </div>
  </div>
  <BackButton />
  <div
    className="double-authorizer-wrapper"
  >
    <MusicCard2
      confirmPlaylist={[Function]}
    >
      <AuthorizerAppleMusic />
      <AuthorizerSpotify />
    </MusicCard2>
  </div>
  <SyncElement
    playlistReady={false}
  />
  <div
    className="how-it-works sync-id"
  >
    <h1
      className="how-it-works-title"
    >
      How to use this sync ID option
    </h1>
    <div
      className="point need"
    >
      <h2
        className="point-title"
      >
        Submit Sync ID
      </h2>
      <div
        className="point-description"
      >
        <p>
          This one is made for when both your peer has already done the second option.
          <br />
          <br />
          Simply, enter the Sync ID that was emailed or texted to you.
          <br />
          <br />
          You then choose the playlist you would like to sync.
          <br />
          <br />
          If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
          <br />
          <br />
          Then, since you peer has already provided their information, you just need to press the Sync button and your playlists will be synced!
          <br />
          <br />
          Enjoy sharing your playlist!
          <br />
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;
